sycl/tools/sycl-ls/sycl-ls.cpp

```diff
@@ -89,6 +90,12 @@ static void printDeviceInfo(const device &Device, bool Verbose,
     for (auto size : sg_sizes)
       std::cout << " " << size;
     std::cout << std::endl;
+
+    auto GraphSupport = Device.get_info<exp_ext::info::device::graph_support>();
+    if (GraphSupport != exp_ext::graph_support_level::unsupported) {
+      std::cout << Prepend << "sycl_ext_oneapi_graph" << std::endl;
+    }
+
```

Comment:
Any chance you can make a proper aspect for this instead? I had a brief chat with @steffenlarsen and it seems that having `aspect::ext_oneapi_graph` if either emulated or native support is present on device would be best here. Then, the query to distinguish between native/emulated could only be present on device that supports the aspect (eliminating unsupported state in it).

Reply:
I'm actually not sure whether the native/emulated query is useful anymore as we've not deemed emulation to have high enough value to implement it yet, so turning this query into an aspect works seems like a sensible change.

Reply:
@aelovikov-intel  @steffenlarsen  I've updated this to be an aspect as you suggested, if you could re-review to see if it lines-up with what you expect. There's a LLVM change outside of the `sycl` directory which I assume is okay to do here, rather than being up-streamed to mainline LLVM then pulled back down, but let me know if not.

Technically I think this is also an ABI break due to the removal of the device info query symbols, but I hope that we can get around that with the experimental status of SYCL-Graphs. If not I can reintroduce the device info query as depreciated in the spec and to be removed later in an ABI break window.


@intel/sycl-graphs-reviewers  should re-review after this change too.